src/Bicep.Core/SourceGraph/ISourceFile.cs

```diff
@@ -13,5 +15,11 @@ public interface ISourceFile
         IFileHandle FileHandle { get; }
 
         string Text { get; }
+
+        RootConfiguration Configuration { get; }
```

Comment:
The intent of this change is to eventually support artifact resolution from ARM template source files using the bicep config. Currently the extension metadata created from ARM template files does not have a namespace type assigned and I left that out of scope for this PR.